icicle-compiler/test/Icicle/Test/Arbitrary/Core.hs

```diff
@@ -178,19 +178,20 @@ instance Arbitrary ValType where
   arbitrary =
    -- Need to be careful about making smaller things.
    -- It's fine if they're big, but they have to fit in memory.
-   oneof_sized_vals
-         [ IntT
-         , UnitT
-         , BoolT
-         , TimeT
-         , StringT ]
+   oneof_sized
+         [ pure IntT
+         , pure UnitT
+         , pure BoolT
+         , pure TimeT
+         , pure StringT
+         , StructT <$> arbitrary
+         , OptionT <$> arbitrary
```

Comment:
the stuff in the "small" list should probably be non-recursive, otherwise you might generate `OptionT (OptionT (OptionT etc` and never terminate